code/game/machinery/doors/door.dm

```diff
@@ -102,6 +102,10 @@ var/list/all_doors = list()
 				playsound(src.loc, 'sound/machines/denied.ogg', 50, 1)
 				door_animate("deny")
 
+/obj/machinery/door/bumped_by_firebird(var/obj/structure/bed/chair/vehicle/wizmobile/W)
+	W.forceMove(get_step(W,W.dir))//Firebird doesn't wait for no slowpoke door to fully open before dashing through!
```

Comment:
You'd need to mess around with `door/airlock/open()` proc, because that thing sleeps first before making itself "undense" (you can pass through it), maybe add a parameter to force-undense it as soon as the proc is ran?

This would basically make the door non-dense FIRST (unlike it does currently, where it sleeps(5) before changing the density) and THEN do the rest of the opening shtick.